docs/build/Build-Script.md

```diff
@@ -130,123 +145,170 @@ If you have multiple link flags, supply it with multiple use of this option;
 
 ```
-./tools/build.py --jerry-memstat
+./tools/build.py --target-os=nuttx --target-arch=arm --target-board=stm32f4dis
 ```
 
 --
-#### `--jerry-lto`
-With given this option, JerryScript will be built with LTO.
+#### `--target-os`
+* `linux` | `darwin` | `osx` | `nuttx` | `tizen` | `tizenrt` | `openwrt`
+
+Specify target OS.
 
 ```
-./tools/build.py --jerry-lto
+./tools/build.py --target-os=nuttx --target-arch=arm
 ```
 
+
+### Arguments of JerryScript
+The following arguments are related to the JavaScript engine under the framework. For example they can change the enabled features of the ECMA-262 standard.
+
 --
-#### `--no-init-submodule`
-With given this option, submoduls will not initialized before start build.
+#### `--jerry-cmake-param`
+Specify CMake parameters for JerryScript.
+
+"cmake" command for JerryScript will be executed with the given parameter applied.
+
+If you have multiple parameters, supply it with multiple use of this option
+
+--
+#### `--jerry-compile-flag`
+Specify C compiler flags for JerryScript.
+
+If you have multiple cflags, supply it with multiple use of this option
 
 ```
-./tools/build.py --no-init-submodule
+./tools/build.py --jerry-compile-flag="-DCONFIG_ECMA_LCACHE_DISABLE"
 ```
 
 --
-#### `--no-check-tidy`
-With given this option, tidy checking will not performed.
+#### `--jerry-debugger`
+Enable JerryScript debugger. See also ["Use JerryScript Debugger"](../devs/Use-JerryScript-Debugger.md).
 
 ```
-./tools/build.py --no-check-tidy
+./tools/build.py --jerry-debugger
 ```
 
 --
-#### `--no-parallel-build`
-With given this option, compilation process will not run in parallel. In other words, executes `make` without `-j` option.
+#### `--jerry-heaplimit`
+Specify object heap limit for JerryScript engine.
 
 ```
-./tools/build.py --no-parallel-build
+./tools/build.py --jerry-heaplimit=80
 ```
 
 --
-#### `--nuttx-home`
-To build for nuttx os, nuttx home directory must be given.
+#### `--jerry-heap-section`
+Specify the name of the JerryScript heap section.
 
 ```
-./tools/build.py --target-os=nuttx --target-arch=arm --target-board=stm32f4dis --nuttx-home="..."
+./tools/build.py --jerry-heaplimit=80
 ```
 
 --
-#### `--run-test`
-With given this option, unit test checking will be performed.
+#### `--jerry-lto`
+With given this option, JerryScript will be built with LTO.
 
 ```
-./tools/build.py --run-test
+./tools/build.py --jerry-lto
 ```
 
 --
-#### `--config`
-Specify build configuration file path.
+#### `--jerry-memstat`
+Enable memstat of JerryScript engine.
 
 ```
-./tools/build.py --config=build.arm.nuttx.stm32f4dis.config
+./tools/build.py --jerry-memstat
 ```
 
-`build.default.config` file is in the source tree for default setting.
+--
+#### `--jerry-profile`
+* `es5.1` | `es2015-subset`
 
-If this option is not specified, `build.config` file will be applied. If the file is not exist, it will be copied from `build.default.config`.
+Specify the profile for JerryScript (default: es5.1).
```

Comment:
Can you add an example of how to specify details of the profiles? e.g) how to selectively enable `arrow function` and `template literals` among the es2015-subsets supported. I think it would be helpful.

Reply:
Hmm, I am a bit surprised about this. I did not know how badly this feature is documented in both IoT.js and JerryScript. Furthermore the `build.py` of IoT.js only supports the built-in profiles. It should work with custom profiles too, but if you give it an absolute path (e.g.: `tools/build.py --clean --jerry-profile=/home/iotjs/custom.profile`), then it does not work. Output: `build.py: error: argument --jerry-profile: invalid choice: '/home/iotjs/custom.profile' (choose from 'es5.1', 'es2015-subset')`

So the build script must be updated as well. I would like to do it in a followup patch if it is good for you.